calico_node/startup/startup.go

```diff
@@ -828,6 +834,68 @@ func ensureNodeFelixConfig(c *client.Client, host, key, def string) error {
 	}
 }
 
+// ensureClusterTypeSet ensures that the ClusterType in the Felix global
+// config has the data contained in the passed in clusterType.  This includes
+// merging what was already set in ClusterType with the passed in clusterType.
+func ensureClusterTypeSet(c *client.Client, key, clusterType string) error {
+	global := ""
+	// Do this twice for the case where we update the field but someone else
+	// overwrites our update, don't do this forever because we don't want to
+	// take too long making this update.
+	for i := 0; i < 2; i++ {
+		if val, assigned, err := c.Config().GetFelixConfig("ClusterType", global); err != nil {
+			return err
+		} else if !assigned {
+			if err := c.Config().SetFelixConfig(key, global, clusterType); err != nil {
+				return err
+			}
+		} else {
+			// Sort so any duplicates will be consecutive
+			all := append(strings.Split(val, ","), strings.Split(clusterType, ",")...)
+			sort.Strings(all)
+
+			if len(all) == 0 {
```

Comment:
Is this ever true?  I think an empty string that is split will return a slice of length 1 containing an empty string.